ops/model.py

```diff
@@ -3613,69 +3603,25 @@ def __init__(
         self._is_leader: bool | None = None
         self._leader_check_time = None
         self._hook_is_running = ''
-        self._is_recursive = contextvars.ContextVar('_is_recursive', default=False)
 
     @contextlib.contextmanager
-    def _prevent_recursion(self):
-        token = self._is_recursive.set(True)
+    def _trace_and_wrap_errors(self, cmd: str, *args: Any, **kwargs: Any):
```

Comment:
I really like how this turned out -- makes the usages so much simpler. However, (my suggestion) of the name is a big of a mouthful. How about just renaming to just `_run` or maybe `_run_hookcmd`? :-)